content/concepts/module-resolution.md

```diff
@@ -3,70 +3,73 @@ title: Module Resolution
+In this case, the directory of the resource file where the `import` or `require` occurs is taken to be the context directory. The relative path specified in the `import/require` is joined to this context path to produce the absolute path to the module.
 
-### Module path
+###  3. Module paths
 
 ```js
 import "module";
 import "module/lib/file";
 ```
 
-Modules are searched for inside directories which are specified using `resolve.modules`, which can be an array comprised of different paths.
-Aliasing, i. e. setting `resolve.alias` to an existing module path, allows you to replace the module path with an alias name during `require/import`.
+Modules are searched for inside all directories specified in [`resolve.modules`](/configuration/resolve/#resolve-modules).
+You can replace the original module path by an alternate path by creating an alias for it using [`resolve.alias`](/configuration/resolve/#resolve-alias) configuration option.
+
+Once the path is resolved based on the above rule, the resolver checks to see if the path points to a file or a directory. If the path points to a file:
+* If the path has a file extension, then the file is bundled straightaway.
+* Otherwise, the file extension is resolved using the [`resolve.extensions`](/configuration/resolve/#resolve-extensions) option, which tells the resolver which extensions (eg - `.js`, `.jsx`) are acceptable for resolution.
 
-Once the path is resolved based on the above rule, the resolver checks if the path points to a file or to a directory. If the path points to a file then it is bundled straightaway.
-But if the path points to a folder, then the following steps are taken to find the right file with the right extension.
-* The right file is determined using the `main: "<filename>.js"` field in `package.json`.
-* If there is no package.json or the main file is not found, `resolve.mainFiles` configuration option is looked up.
-* If this also fails, then it looks for a file named `index` by default.
-* `resolve.extensions` tells the resolver which extensions (eg - `.js, .jsx`) are acceptable for resolution.
+If the path points to a folder, then the following steps are taken to find the right file with the right extension:
+* If the folder contains a `package.json` file, then fields specified in [`resolve.mainFiles`](/configuration/resolve/#resolve-mainfields) configuration option are looked up in order, and the first such field in `package.json` determines the file path. 
+* If there is no `package.json` or if the main fields do not return a valid path, file names specified in the [`resolve.mainFiles`](/configuration/resolve/#resolve-mainfiles) configuration option are looked for in order, to see if a matching filename exists in the imported/required directory .
+* The file extension is then resolved in a similar way using the `resolve.extensions` option.
+
+Webpack provides reasonable [defaults](/configuration/resolve) for these options depending on your build target.
```

Comment:
Explained module resolution in detail, making no assumption of `target`. e.g. `main` field in `package.json` is searched if target is node, else browser is searched.